lib/base-controller.js

```diff
@@ -23,6 +23,41 @@ function getErrorLength() {
   }
 }
 
+/*
+ * Utility function which returns the
+ * name of the step a field is present on
+ */
+function getStepFromFieldName(fieldName, steps) {
+  return _.findKey(steps, function findKey(step) {
+    if (!step.fields) {
+      return false;
+    }
+    return step.fields.indexOf(fieldName) > -1;
+  });
+}
+
+/*
+ * Utility function to create an array
+ * of field objects made up of field name,
+ * value and step
+ */
+function getFieldObjectsFromNames(fields, values, modifiers, steps) {
+  return fields.filter(function filter(item) {
+    return (values[item] !== undefined && values[item] !== '') || modifiers[item] !== undefined;
+  }).map(function map(item) {
+    var value = values[item];
+    var step = getStepFromFieldName(item, steps);
+    if (modifiers && modifiers[item]) {
```

Comment:
Presumably `modifiiers` is already known to be defined because line 46 potentially executes that test